lib/samples/apply_symbology_to_shapefile/apply_symbology_to_shapefile.dart

```diff
@@ -0,0 +1,146 @@
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                  children: [
+                    const Text('Alternate renderer'),
+                    Switch(
+                      value: _alternate,
+                      onChanged: updateRenderer,
+                    ),
+                  ],
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a map with the topographic basemap style and an initial viewpoint.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    map.initialViewpoint = Viewpoint.fromCenter(
+      ArcGISPoint(
+        x: -11662054,
+        y: 4818336,
+        spatialReference: SpatialReference(wkid: 3857),
+      ),
+      scale: 200000,
+    );
+
+    // Download the sample data.
+    await downloadSampleData(['d98b3e5293834c5f852f13c569930caa']);
+    // Get the application documents directory.
+    final appDir = await getApplicationDocumentsDirectory();
+    // Get the Shapefile from the download resource.
+    final shapefile =
+        File('${appDir.absolute.path}/Aurora_CO_shp/Subdivisions.shp');
+    // Create a feature table from the Shapefile URI.
+    final shapefileFeatureTable =
+        ShapefileFeatureTable.withFileUri(shapefile.uri);
+    // Create a feature layer for the Shapefile feature table.
+    _shapefileFeatureLayer =
+        FeatureLayer.withFeatureTable(shapefileFeatureTable);
+    // Clear the operational layers and add the feature layer to the map.
+    map.operationalLayers.clear();
+    map.operationalLayers.add(_shapefileFeatureLayer);
+    // Create the symbology for the alternate renderer.
+    final lineSymbol = SimpleLineSymbol(
+      style: SimpleLineSymbolStyle.solid,
+      color: Colors.red,
+      width: 1.0,
+    );
+    final fillSymbol = SimpleFillSymbol(
+      style: SimpleFillSymbolStyle.solid,
+      color: Colors.yellow,
+      outline: lineSymbol,
+    );
+    // Create the alternate renderer.
+    _alternateRenderer = SimpleRenderer(symbol: fillSymbol);
+    // Wait for the layer to load so that it will be assigned a default renderer.
+    await _shapefileFeatureLayer.load();
+    // Hold a reference to the default renderer (to enable switching between the renderers).
+    _defaultRenderer = _shapefileFeatureLayer.renderer;
+
+    _mapViewController.arcGISMap = map;
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  // Set the renderer.
+  void updateRenderer(bool value) {
+    setState(() => _alternate = value);
+    if (_shapefileFeatureLayer.renderer == _defaultRenderer) {
```

Comment:
I would change this to just `if (_alternate) {`. That way we're guaranteed that the renderer is in sync with the value of `_alternate`.

Also instead of naming the parameter `value`, it should have a more descriptive name, like `alternate`.